Build errors on macOS 10.12 Sierra: clock_gettime

I'm not able to build civetweb on the macOS 10.12 Sierra public beta. Apple has declared & defined clock_gettime starting in 10.12, so this code causes build errors:

[ 98%] Building C object CMakeFiles/src/lib/civetweb/civetweb.c.o
src/lib/civetweb/civetweb.c:135:5: error: conflicting types for 'clock_gettime'
int clock_gettime(int clk_id, struct timespec *t);
    ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.12.sdk/usr/include/time.h:177:5: note: 
      previous declaration is here
int clock_gettime(clockid_t __clock_id, struct timespec *__tp);
    ^
src/lib/civetweb/civetweb.c:138:1: error: conflicting types for 'clock_gettime'
clock_gettime(int clk_id, struct timespec *t)
^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.12.sdk/usr/include/time.h:177:5: note: 
      previous declaration is here
int clock_gettime(clockid_t __clock_id, struct timespec *__tp);
    ^
2 errors generated.
make[2]: *** [CMakeFiles/src/lib/civetweb/civetweb.c.o] Error 1
make[2]: *** Waiting for unfinished jobs....


See similar reports in other projects:
• https://github.com/Homebrew/homebrew-x11/issues/219
• https://github.com/jordansissel/xdotool/issues/138

--- a/src/civetweb.c
+++ b/src/civetweb.c
@@ -136,8 +136,17 @@ mg_static_assert(sizeof(void *) >= sizeof(int), "data type size check");
 #include <mach/mach_time.h>
 #include <assert.h>
 
+/* Determine if the current OSX version supports clock_gettime */
+#ifdef __APPLE__
+  #include <AvailabilityMacros.h>
+  #ifndef MAC_OS_X_VERSION_10_12
+    #define MAC_OS_X_VERSION_10_12 101200
+  #endif
+#endif
+#define CIVETWEB_APPLE_HAVE_CLOCK_GETTIME defined(__APPLE__) && MAC_OS_X_VERSION_MIN_REQUIRED >= MAC_OS_X_VERSION_10_12
 
-/* clock_gettime is not implemented on OSX */
+#if !(CIVETWEB_APPLE_HAVE_CLOCK_GETTIME)
+/* clock_gettime is not implemented on OSX prior to 10.12 */
 int clock_gettime(int clk_id, struct timespec *t);
 
 int
@@ -181,6 +190,7 @@ clock_gettime(int clk_id, struct timespec *t)
 	}
 	return -1; /* EINVAL - Clock ID is unknown */
 }
+#endif
 #endif
 
 

